src/FormField.js

```diff
@@ -83,33 +83,49 @@ class FormField extends React.Component {
     let fieldPlaceholder
     let iconAdjustment
 
-    React.Children.forEach(children, (child, index) => {
-      if (!child) {
-        return
-      }
+    const parseLabel = child => {
+      LabelChild = child
+    }
 
-      const { type, props } = child
+    const parseInputOrSelect = (child, props, index) => {
+      position = index
+      FieldChild = child
+      fieldId = props.id
+      // For aria-label when Label child is not rendered
+      fieldPlaceholder = props.placeholder
+    }
 
-      if (type === Label) {
-        LabelChild = child
+    const parseIcon = (child, props) => {
+      if (position < 0) {
+        BeforeIcon = child
+        iconAdjustment = props.size - 24
+      } else {
+        AfterIcon = child
       }
+    }
 
-      if (type === Input || type === Select) {
-        position = index
-        FieldChild = child
-        fieldId = props.id
-        // For aria-label when Label child is not rendered
-        fieldPlaceholder = props.placeholder
-      }
+    const parseChild = (child, index) => {
+      const { type, props } = child
 
-      if (type === Icon) {
-        if (position < 0) {
-          BeforeIcon = child
-          iconAdjustment = props.size - 24
-        } else {
-          AfterIcon = child
-        }
+      switch (type) {
+        case Label:
+          parseLabel(child)
+          break
+        case Input:
+          parseInputOrSelect(child, props, index)
+          break
+        case Select:
```

Comment:
Since these two blocks are almost the same, couldn't this be written as:

```js
case Input:
case Select:
  parseInputOrSelect(child, props, index)
  break
```